### DISCUSSION TITLE
53. IP 주소를 두 파트로 나누어 설명해주세요(with 서브넷 마스크).

### DISCUSSION THREAD

**Discussion by linirini** [jewelry-box/cs-interview#55] [General]
- 그리고 서브넷 마스크를 설명해주세요.

`주요 키워드`: IP주소, 네트워크 주소, 호스트 주소, 서브넷 마스크
`추가 키워드`: 서브넷, CIDR 표기법

**Comment by cutehumanS2** [comment]
### IP 주소를 두 파트로 나누어 설명해 주세요.
IP 주소를 두 파트로 나누어 설명하자면, **네트워크 주소**와 **호스트 주소**로 나눌 수 있습니다. 
- 네트워크 주소는 호스트가 속한 특정 네트워크를 식별하는 역할을 하며, 같은 네트워크에 속한 장치들은 같은 네트워크 주소를 갖습니다.
- 호스트 주소는 네트워크 내에서 특정 호스트를 식별하는 역할을 합니다. 네트워크에 속한 각 장치마다 고유한 값이 할당됩니다.

<hr>

### 서브넷 마스크란 무엇인가요? 어떻게 사용되나요?
서브넷 마스크는 IP 주소에서 네트워크 주소와 호스트 주소를 구분짓는 비트열입니다. 32비트로 구성되며, 네트워크 영역은 1, 호스트 영역은 0으로 표기합니다. 

IP 주소와 서브넷 마스크를 비트 AND 연산하여, 네트워크 주소를 구하는 데 사용됩니다.

**Comment by cutehumanS2** [comment]
### 추가 질문

**[서브넷이 무엇이며, 왜 필요한가요?]**

서브넷은 하나의 네트워크를 더 작은 네트워크를 나눈 것입니다. 이를 통해 IP 주소를 더 효율적으로 사용할 수 있습니다.

기존의 클래스 기반 주소 체계에서는 네트워크 크기가 고정되어 있어 IP 주소 낭비가 발생할 가능성이 높았지만, 서브넷을 사용하면 필요한 만큼의 호스트 수에 맞게 네트워크를 구성할 수 있습니다. 

예를 들어, 300명의 직원이 사용할 컴퓨터들을 동일한 네트워크로 구성하고 싶을 때, 서브네팅을 사용하면 C 클래스의 기본 네트워크 범위를 확장해 낭비 없이 사용할 수 있습니다. 
<br><br>

**[서브넷 마스크와 CIDR 표기법의 차이점은 무엇인가요?]**

- `IP 주소/서브넷 마스크상의 1의 개수` 형식으로 표기하는 방법 → CIDR 표기법
- 서브넷 마스크는 10진수 형식으로 IP 주소처럼 4개의 옥텟(8비트)으로 표현됩니다. 예) 255.255.255.0
- CIDR 표기법은 `IP 주소/서브넷 마스크상의 1의 개수` 형식으로 표기하여, 서브넷 마스크를 더 간결하게 표현할 수 있습니다.
    
    +) Classless Inter-Domain Routing(클래스없는 도메인 간 라우팅)
    
<br><br>

[참고한 자료]

<혼자 공부하는 네트워크> - 03-2. IP 주소
